docs/custom_tool_guides/slack.md

```diff
@@ -0,0 +1,128 @@
+# Slack Tool Setup
+
+To set up the Slack tool you will need a Slack application. Follow the steps below to set it up:
+
+## 1. Create a Slack App
+
+Head to the [Slack API](https://api.slack.com/apps) and create a new app.
+After creating the app, you will see the `App Credentials` section. Copy the `Client ID` and `Client Secret` values.
+That will be used for the environment variables specified above.
+
+## 2. Set up OAuth & Permissions
+OAuth flow is required to authenticate users with Slack. 
+To enable it please set the following redirect URL to your app's settings:
+```bash
+    https://<your_backend_url>/v1/tool/auth
+```
+Please note that for the local development you will need to enable HTTPS. 
+See the [Setup HTTPS for Local Development](#5-setup-https-for-local-development) section for more details.
+If you are using a local https setup, redirect url should be 
+```
+ https://localhost:8000/v1/tool/auth
+```
+Also, you can set up a proxy, such as [ngrok](https://ngrok.com/docs/getting-started/), to expose your local server to the internet.
+
+The Slack tool uses User Token Scopes to access the user's Slack workspace.
+The required and the default permission scope is `search:read`.
+Set it in the `OAuth & Permissions` section of your Slack app settings.
+
+To work with the Slack Tool Advanced token security via token rotation is required.
+To enable it, go to the `OAuth & Permissions` section of your Slack app settings and click 'Opt in' button in the 'Advanced token security via token rotation' section.
+
+More information about the OAuth flow can be found [here](https://api.slack.com/authentication/oauth-v2).
+
+## 3. Set Up Environment Variables
+
+Then set the following environment variables. You can either set the below values in your `.env` file:
+
```

Comment:
Small request: Can you switch the ordering of 3. so that we recommend updating `secrets.yaml` first, then offer `.env` as a fallback solution

Reply:
Thanks, I switched the order